File: js/baekjoon/step4/bkjn_10811_1.js
const input = require("fs")
  .readFileSync("dev/stdin")
  .toString()
  .trim()
  .split("\n");

const [n, m] = input[0].split(" ").map(Number);
let baskets = Array(n)
  .fill()
  .map((v, i) => i + 1);

for (j = 1; j <= m; j++) {
  let [a, b] = input[j].split(" ").map(Number);
  let tmp = [];
  for (k = a - 1; k < b; k++) {
    tmp.push(baskets[k]);
  }
  tmp.reverse();
  baskets.splice(a - 1, b - a + 1, ...tmp);
}
console.log(baskets.join(" "));

File: python/baekjoon/bkjn_10811_1.py
n, m = map(int, input().split())
basket = [i for i in range(1, n + 1)]

for _ in range(m):
  i, j = map(int, input().split())
  arr = basket[i-1:j]
  arr.reverse()
  basket[i-1:j] = arr

for i in range(n):
  print(basket[i], end=" ")